website/server/controllers/api-v3/groups.js

```diff
@@ -348,7 +348,13 @@ api.leaveGroup = {
     if (validationErrors) throw validationErrors;
 
     let group = await Group.getGroup({user, groupId: req.params.groupId, fields: '-chat', requireMembership: true});
-    if (!group) throw new NotFound(res.t('groupNotFound'));
+    if (!group) {
+      delete user.party._id;
```

Comment:
Seems to me like this would remove a user from their party of the guild they tried to leave didn't exist. This should do a check first to see if the party id equals the id in question. If it does, delete it. If it doesn't, remove it from the guilds array.